worldwind/src/commonMain/kotlin/earth/worldwind/shape/Placemark.kt

```diff
@@ -443,17 +442,58 @@ open class Placemark @JvmOverloads constructor(
      * @param rc       the current render context
      * @param drawable the Drawable to be prepared
      */
-    protected open fun prepareDrawableLeader(rc: RenderContext, drawable: DrawableLines) {
+    protected open fun prepareDrawableLeader(rc: RenderContext, drawable: DrawableDynamicLines) {
         // Use the basic GLSL program to draw the placemark's leader.
-        drawable.program = rc.getShaderProgram { BasicShaderProgram() }
-
-        // Compute the drawable's vertex points, in Cartesian coordinates relative to the placemark's ground point.
-        drawable.vertexPoints[0] = 0f // groundPoint.x - groundPoint.x
-        drawable.vertexPoints[1] = 0f // groundPoint.y - groundPoint.y
-        drawable.vertexPoints[2] = 0f // groundPoint.z - groundPoint.z
-        drawable.vertexPoints[3] = (placePoint.x - groundPoint.x).toFloat()
-        drawable.vertexPoints[4] = (placePoint.y - groundPoint.y).toFloat()
-        drawable.vertexPoints[5] = (placePoint.z - groundPoint.z).toFloat()
+        drawable.program = rc.getShaderProgram { GeomLinesShaderProgram() }
+
+        var vertexIndex = 0
+        drawable.vertexPoints[vertexIndex++] = (placePoint.x - groundPoint.x).toFloat()
```

Comment:
I see similar code in multiple places - any chance for a common function to generate 1 line vertex?